### PR TITLE
Disable babel, improve test speed, use Chrome headless...

### PR DESCRIPTION
**Pull request by vperron**
# What's done
- [x] Do not transpile if target is a library (or according to a setting)
- [x] Get rid of PhantomJS and use Chrome Headless
- [x] Use "eval" devtool in tests to speedup builds
- [x] Speedup builds using HardSource plugin
- [x] Fixups and dependency updates

Having libraries distributed as ES6 native bundles enables tremendous bundle savings after optimized passes (~30% in the case of the tested app, without looking much further)

# Timings
Timings about "lookcoco" app. J'ai essayé toutes les méthodes imaginables.
```
[vperron@M]<~/dev/lookcoco> time make test                                                                     
mkdir -p dist/
gettext-compile --output dist/translations.json locale/en_GB/LC_MESSAGES/app.po locale/fr_FR/LC_MESSAGES/app.po
[gettext] processing PO file: locale/en_GB/LC_MESSAGES/app.po
[gettext] processing PO file: locale/fr_FR/LC_MESSAGES/app.po
(node:13417) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
mkdir -p dist/
--- [make] generating 'app.settings.js' 
./node_modules/.bin/ini2js src/settings/00_common.example.ini --global_name __SETTINGS__ > dist/app.settings.js
eslint --config ./node_modules/systematic/.eslintrc --ext .js,.vue --plugin html src
karma start --reporters webpack-error --single-run --no-auto-watch karma.conf.js

START:
26 10 2017 11:00:09.351:INFO [karma]: Karma v1.7.1 server started at http://0.0.0.0:9876/
26 10 2017 11:00:09.353:INFO [launcher]: Launching browser PhantomJS with unlimited concurrency
26 10 2017 11:00:09.381:INFO [launcher]: Starting browser PhantomJS
26 10 2017 11:00:10.408:INFO [PhantomJS 2.1.1 (Linux 0.0.0)]: Connected on socket vapVeioyUvhoDv5pAAAA with id 91288614
  ErrorSnackbar.spec.js
    ✔ should detect and clear error
  StationsRow.spec.js
    ✔ should display a row
  station.spec.js
    ✔ should count the station according to their state
    ✔ should change the ordering of files
  Stations.spec.js
    ✔ should mount the page of Stations

Finished in 0.041 secs / 0.219 secs @ 11:00:11 GMT+0200 (CEST)

SUMMARY:
✔ 5 tests completed

real    0m15.032s
user    0m16.912s
sys     0m0.700s
```

# Result Table
- Chrome Headless: 0m12.392s (3s improvement)
- Source-Map as "eval": 0m9.041s (3s improvement)
- Hard Source Plugin: 0m7.917s (2s improvement)
- noParse sur Vuetify: 0m7.035s (1s improvement, mais ça casse vite sur d'autres libs)
- Stop checking syntax + prepare before tests: 0m3.819s (3s improvement)

- Without Babel: aucun effet, dans aucun des cas vérifiés; et dangereux.
- DllPlugin (bundler node_modules en avance): très complexe et aucun effet. Ccasse vanitils ES6.
- HappyPack (webpack multithread): aucun effet.
- node --perf & node --inspect-brk + Chrome: aucun axe d'amélioration, cf. image.

# Finally on lookcoco
... and we could still not parse Vuetify, which accounts for ~2s of extra compilation.
```
[vperron@M]<~/dev/lookcoco> time PATH=$HOME/bin/chrome-linux:$PATH make test
Makefile:13: warning: overriding recipe for target '/tmp/template.pot'
node_modules/systematic/mk/main.mk:186: warning: ignoring old recipe for target '/tmp/template.pot'
karma start --reporters webpack-error --single-run --no-auto-watch karma.conf.js

START:

02 11 2017 18:06:05.177:INFO [karma]: Karma v1.7.1 server started at http://0.0.0.0:9876/
02 11 2017 18:06:05.179:INFO [launcher]: Launching browser ChromeHeadless with unlimited concurrency
02 11 2017 18:06:05.203:INFO [launcher]: Starting browser ChromeHeadless
02 11 2017 18:06:05.536:INFO [HeadlessChrome 0.0.0 (Linux 0.0.0)]: Connected on socket Xsc8hqV95m5aJIAGAAAA with id 68297049
HeadlessChrome 0.0.0 (Linux 0.0.0) INFO: 'Download the Vue Devtools extension for a better development experience:
https://github.com/vuejs/vue-devtools'
  ErrorSnackbar.spec.js
    ✔ should detect and clear error
  StationRowMaintenance.spec.js
    ✔ should display a row
  equipment.spec.js
    ✔ should extract the name and the kind of the equipement
  station.spec.js
    with a large set of data
      ✔ should count the station according to their state
      ✔ should change the ordering of files
INFO: 'Download the Vue Devtools extension for a better development experience:
https://github.com/vuejs/vue-devtools'
INFO: 'Download the Vue Devtools extension for a better development experience:
https://github.com/vuejs/vue-devtools'
    ✔ should filter files according the page
  Stations.spec.js
    ✔ should mount the page of Stations

Finished in 0.29 secs / 0.287 secs @ 18:06:06 GMT+0100 (CET)

SUMMARY:
✔ 7 tests completed

real    0m5.115s
user    0m4.524s
sys     0m0.448s

```

# Flame graph
screenshot from 2017-11-02 14-30-29

### CODE REVIEW

File: js/config.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ defaultsDeep(systematic, {
     output_dir: 'dist',
     public_path: '/',
     enable_filename_hashing: true,
+    enable_babel: true,
+    es_libs: [],
```
Comment (cedricco) [inline js/config.js]: The project needs to declare its ES6 modules libs ?

File: mk/main.mk [hunk 2]
```diff
@@ -140,7 +140,7 @@ syntax: eslint
 eslint:
 	eslint --config $(ESLINTRC) $(ESLINTOPTIONS) $(SRC_DIR)
 
-test: prepare syntax
+test:
```
Comment (cedricco) [inline mk/main.mk]: Mandatory syntax checking was indeed annoying :)

File: mk/main.mk [hunk 3]
```diff
@@ -164,10 +164,14 @@ serve: prepare
...
 # Don't minify because it causes issues, see https://github.com/Polyconseil/systematic/issues/13
 dist: prepare
+ifeq ($(BUILD_TYPE),library)
+	@echo "Doing nothing, it's a library. Export the entry point in your package.json!"
+else
 	mkdir -p $(OUTPUT_DIR)
 	NODE_ENV=production webpack $(WEBPACK_OPTIONS_CONFIG_FILE) \
 		--no-color --bail --display-modules \
 		$(WEBPACK_DIST_OPTS) $(WEBPACK_EXTRA_OPTIONS)
+endif
```
Comment (cedricco) [inline mk/main.mk]: makes me think of linking a library with `npm link`. To still be able to do that, my understanding is that you will have to change `package.json`'s entry point in that library. Is that correct ?

File: js/config.js (hunk 1 above)
Comment (vperron) [reply js/config.js]: yep. Fortunately there's not many. It's either this or we can... take advantage of the "module" field in those dependencies package.json. I'll do this.

File: mk/main.mk (hunk 3 above)
Comment (vperron) [reply mk/main.mk]: Yep, you need to point the "main" entry to the source index of your library.

**Pull request merged**